presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp

```diff
@@ -242,11 +243,56 @@ std::string PrestoServerOperations::taskOperation(
 
 std::string PrestoServerOperations::serverOperation(
     const ServerOperation& op,
-    proxygen::HTTPMessage* /* unused */) {
+    proxygen::HTTPMessage* message) {
   switch (op.action) {
-    case ServerOperation::Action::kTrace: {
+    case ServerOperation::Action::kTrace:
       return velox::process::TraceContext::statusLine();
-    }
+    case ServerOperation::Action::kSetState:
+      if (server_) {
```

Comment:
nit: can we have a dispatch method to handle this? Thanks!

Reply:
You mean a method per action?
Sure, we can do that.